### DISCUSSION TITLE
WASM as Azure Static Web App : 15000 file limit

### DISCUSSION THREAD

**Discussion by baskren** [unoplatform/uno#16326] [Q&A] [answered]
So, [each Azure Static Web Apps - both paid and free - has a 15,000 files limit](https://learn.microsoft.com/en-us/azure/static-web-apps/quotas):
<img width="975" alt="image" src="https://github.com/unoplatform/uno/assets/2528888/1faac5dd-2336-45b6-8892-135b19233401">

In that past that has been no concern for my WASM apps.  However recently I've failed to deployed ... because of this limit.  [This GitHub issue](https://github.com/Azure/static-web-apps/issues/782#issuecomment-1186322614) proposes deleting the  `node_modules` folder (it has 19,316 files).  Sure enough, it seems to work.

So my question is : *Is this a legitimate work around?* 
- If no, why (for my understanding)?
- If yes, could deleting this folder be added to the WASM build process?

**Comment by jeromelaban** [comment, marked as answer]
Thanks for the report. I'm not sure how the node_modules could end up in your output folder. Could you share a scenario where that happens?

**Reply by baskren** [reply]
... oh, and thanks for the prompt response.

**Comment by baskren** [comment]
@jeromelaban  : 

1.  I guess that answers my question!  Thanks!
2.  OK, I'm the fool.  I'm pretty sure I had `rm -rf **/bin` and `rm -rf **/obj`  the entire solution before building and yet, I can't replicate the issue this morning.  

If I am able to replicate it again, would you be ok if I shared a private repo with instructions on how to replicate?

**Reply by jeromelaban** [reply]
It's generally best to avoid private repos, but let us know if you're able to replicate anyways!

**Comment by baskren** [comment]
OK, figured out what caused the appearance of the `node_modules` folder:

- `cd` to the `$(ProjectName).Wasm/bin/Release/net8.0/dist` folder
- Verified that there was not a `node_modules` folder.
- Started to upload the build to Azure Static Web Apps via `swa deploy ../dist --env production`
- Upload terminated without indication of success or failure.  Instead, it gave the following message:
```text
   ╭────────────────────────────────────────────────────╮
   │                                                    │
   │          Update available 1.1.6 → 1.1.7            │
   │   Run npm i @azure/static-web-apps-cli to update   │
   │                                                    │
   ╰────────────────────────────────────────────────────╯
```
- Ran the above command

Sure enough, this is what created the `node_modules` folder.  

Thanks for your patience with me.
